File: 13. Roman to Integer/roman-to-integer.c
// 13. Roman to Integer (1/27/54192)
// Runtime: 4 ms (85.70%) Memory: 5.97 MB (22.73%) 

int romanToInt (char* s) {
    int dict[128] = { 0 };
    dict['I'] = 1;
    dict['V'] = 5;
    dict['X'] = 10;
    dict['L'] = 50;
    dict['C'] = 100;
    dict['D'] = 500;
    dict['M'] = 1000;
    
    int len = strlen(s);
    int ans = dict[s[len - 1]];
    
    for (int i = 0; i < len - 1; i++) {
        if (dict[s[i]] >= dict[s[i + 1]]) {
            ans += dict[s[i]];
        } else {
            ans -= dict[s[i]];
        }
    }
    
    return ans;
}

File: 13. Roman to Integer/roman-to-integer.ts
// 13. Roman to Integer (1/21/54192)
// Runtime: 230 ms (24.32%) Memory: 48.64 MB (42.38%) 

function romanToInt(s: string): number {
    const dict = { M: 1000, D: 500, C: 100, L: 50, X: 10, V: 5, I: 1 };
    
    let ans = dict[s[s.length - 1]];
    
    for (let i = 0; i < s.length - 1; i++) {
        if (dict[s[i]] < dict[s[i + 1]]) {
            ans -= dict[s[i]];
        } else {
            ans += dict[s[i]];
        }
    }
    
    return ans;
};
